portfolio/src/main/java/com/google/sps/data/CalendarGroup.java

```diff
@@ -0,0 +1,126 @@
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/** This class models a group of calendar events. */
+public class CalendarGroup {
+  public List<CalendarEvent> events;
+  public List<TimeRange> allFreeTimeRanges;
```

Comment:
Is this variable sorted? Consider adding comment and ensuring that it is.

That being said, I'm not sure whether you need this variable, because it can be computed based on events and overall start/end times. So Maybe remove this variable, and just expose a method that returns all free time ranges.